reactive/flannel.py

```diff
@@ -141,6 +142,12 @@ def reconfigure_flannel_service():
     remove_state('flannel.service.installed')
 
 
+@when('etcd.available')
+def etcd_changed(etcd):
+    if data_changed('flannel_etcd_connections', etcd.get_connection_string()):
```

Comment:
`data_changed` is hilarious in that the first time it runs, it will always be true.  So as written, we'll do the following on our first run through reactive:

- render our template and set the `installed` flag in `install_flannel_service`
- remove the `installed` flag in this `etcd_changed` handler because d_c is true
-  jump back to `install_flannel_service`, re-rendering the template and re-setting the `installed` flag
- fall into this `etcd_changed` handler again, which is a noop because d_c will be false this time

To prevent the double run of `install_flannel_service` and `etcd_changed`, you could prime the data_changed key in `install_flannel_service`. Something like:

```python
 def install_flannel_service(etcd):
     ''' Install the flannel service. '''
     status_set('maintenance', 'Installing flannel service.')
+    # keep track of our etcd conn string so we can detect when it changes later
+    data_changed('flannel_etcd_connections', etcd.get_connection_string())
     iface = config('iface') or get_bind_address_interface()
```